core/src/main/java/org/apache/brooklyn/util/core/file/ArchiveUtils.java

```diff
@@ -363,21 +363,38 @@ public static String readFullyString(File sourceFile) {
     }
 
     public static void extractZip(final ZipFile zip, final String targetFolder) {
-        new File(targetFolder).mkdir();
+        File targetPath = new File(targetFolder);
+        targetPath.mkdir();
         Enumeration<? extends ZipEntry> zipFileEntries = zip.entries();
         while (zipFileEntries.hasMoreElements()) {
-            ZipEntry entry = (ZipEntry) zipFileEntries.nextElement();
-            File destFile = new File(targetFolder, entry.getName());
+            ZipEntry entry = zipFileEntries.nextElement();
+            String originalName = entry.getName();
+            File destFile = new File(targetPath, originalName);
+            // validate input path
+            try {
+                String canonicalDestinationDirPath = null;
+                // enforce the file uses the appropriate file separator for the controller SO
+                String sanitizedName = originalName
+                        .replace("\\", File.separator)
+                        .replace("/", File.separator);
+                File sanitizedDestFile = new File(targetPath, sanitizedName);
+                canonicalDestinationDirPath = targetPath.getCanonicalPath();
```

Comment:
i thought `getCanonicalPath()` didn't work unless the path existed.  but a quick test reveals i was wrong.  looks good.

Reply:
Yeah, it work and it also resolve symbolic links that are also a risk